zircon.swing/src/main/kotlin/org/codetome/zircon/internal/terminal/SwingTerminalFrame.kt

```diff
@@ -47,13 +48,18 @@ class SwingTerminalFrame(title: String = "ZirconTerminal",
         canvas.createBufferStrategy(2)
         swingTerminal.initializeBufferStrategy()
         addWindowStateListener(this)
+        TerminalCanvas::class.javaObjectType.cast(canvas).onPaint = { swingTerminal.flush() }
     }
 
     override fun close() {
         dispose()
     }
 
-    companion object {
-        private fun createCanvas() = Canvas()
+    private class TerminalCanvas() : Canvas() {
+        var onPaint: (() -> Unit)? = null
```

Comment:
You can use `lateinit var onPaint: (() -> Unit)` here so you don't have to bother with the `null`s